### PR TITLE
feat: Introduce Automation API client from -core library

### PR DESCRIPTION
**Pull request by UnseenWizzard**
#### What this PR does / Why we need it:
Replace the monaco client implementation with the new client from the library.

#### Special notes for your reviewer:
There's also small improvements to E2E test utils and delete error handling that I noticed while getting the new client to work with monaco.
Review per commit recommended as always.

#### Does this PR introduce a user-facing change?
Tangential change: Bucket deletion will no longer report 404 responses as an error (and thus behave like other deletions that leniently accept a 404 as the desired state is reached anyway)

### CODE REVIEW

File: pkg/client/clientset.go [hunk 1]
```diff
@@ -179,6 +171,11 @@ func CreatePlatformClientSet(url string, auth PlatformAuth, opts ClientOptions)
 		return nil, err
 	}
 
+	autClient, err := clientFactory.AutomationClient()
```
Comment (UnseenWizzard) [inline pkg/client/clientset.go:174]: setup is reused from the clientFactory built for the bucket client, which is nice and simple 👌

File: pkg/delete/delete.go [hunk 2]
```diff
@@ -385,10 +395,13 @@ func AllAutomations(ctx context.Context, c automationClient) []error {
 		log.WithCtxFields(ctx).WithFields(field.Type(string(resource))).Info("Deleting %d objects of type %q...", len(objects), resource)
 		for _, o := range objects {
 			log.WithCtxFields(ctx).WithFields(field.Type(string(resource)), field.F("object", o)).Debug("Deleting Automation object with id %q...", o.ID)
-			err = c.Delete(t, o.ID)
+			resp, err := c.Delete(ctx, t, o.ID)
 			if err != nil {
 				errs = append(errs, err)
 			}
+			if err, isErr := resp.AsAPIError(); isErr {
```
Comment (UnseenWizzard) [inline pkg/delete/delete.go:402]: note to self: include 404 being fine here as well (line 206)

File: pkg/delete/delete.go [hunk 3]
```diff
@@ -385,10 +395,13 @@ func AllAutomations(ctx context.Context, c automationClient) []error {
 		log.WithCtxFields(ctx).WithFields(field.Type(string(resource))).Info("Deleting %d objects of type %q...", len(objects), resource)
 		for _, o := range objects {
 			log.WithCtxFields(ctx).WithFields(field.Type(string(resource)), field.F("object", o)).Debug("Deleting Automation object with id %q...", o.ID)
-			err = c.Delete(t, o.ID)
+			resp, err := c.Delete(ctx, t, o.ID)
 			if err != nil {
 				errs = append(errs, err)
 			}
+			if err, isErr := resp.AsAPIError(); isErr && resp.StatusCode != http.StatusNotFound { // 404 means it's gone already anyway
```
Comment (warber) [inline pkg/delete/delete.go:402]: the error is added twice then, no?

Comment (warber) [reply pkg/delete/delete.go:402]: ah no, fortget it, thy are different of course

File: internal/automationutils/response_decoder.go
Comment (warber) [inline internal/automationutils/response_decoder.go:1]: maybe add tests for this?

**Review by warber — DISMISSED** [review]

**Review by warber — APPROVED** [review]

**Pull request merged**